store/postgres/tests/store.rs

```diff
@@ -28,15 +29,24 @@ fn postgres_test_url() -> String {
         .unwrap()
 }
 
+lazy_static! {
+    static ref TEST_MUTEX: Mutex<()> = Mutex::new(());
+}
+
 /// Test harness for running database integration tests.
 fn run_test<T>(test: T) -> ()
 where
-    T: FnOnce() -> () + panic::UnwindSafe,
+    T: FnOnce() -> () + Send + 'static + panic::UnwindSafe,
 {
-    insert_test_data();
-    let result = panic::catch_unwind(|| test());
-    remove_test_data();
-    result.expect("Failed to run test");
+    let _test_lock = TEST_MUTEX.lock().unwrap();
+    tokio::run(future::lazy(|| {
```

Comment:
I am noticing tests pass that should be failing.  Example: If I have not set `THEGRAPH_STORE_POSTGRES_DIESEL_URL` all tests should fail, but are currently passing.

Reply:
I also tested the changes by using a valid `THEGRAPH_STORE_POSTGRES_DIESEL_URL`, but changing an assertion to force a test failure. Again all tests passed.  In both cases the process did log a panic, but the test did not fail.

Reply:
Great catch!